share/p2p/shrexeds/client.go

```diff
@@ -143,11 +147,13 @@ func (c *Client) doRequest(
 func (c *Client) setStreamDeadlines(ctx context.Context, stream network.Stream) {
 	// set read/write deadline to use context deadline if it exists
 	if dl, ok := ctx.Deadline(); ok {
-		err := stream.SetDeadline(dl)
-		if err == nil {
-			return
-		}
-		log.Debugw("client: setting deadline: %s", "err", err)
+		stream.SetDeadline(dl) //nolint:errcheck
+		// TODO(@distractedm1nd): This is commented out as part of a hotfix until we find good defaults for deadlines.
+		// particularly, we can get stuck reading the status from the stream for 60s if we don't comment this out
+		// if err == nil {
+		//	return
+		// }
+		// log.Debugw("client: setting deadline: %s", "err", err)
```

Comment:
I don't get this change, it looks like you commented out logging an error of deadline is not set.

Reply:
The commenting out is to avoid the early return, and also avoid the log after (which should only be there on err).

This means we set the ReadDeadline to 5s (or whatever is in the config), which fixes the second context deadline (reading status from stream)